packages/one-app-hmr/lib/webpack/plugins/hot-holocron-loader.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import loaderUtils from 'loader-utils';
+
+const modify = (src, { moduleName }) => {
+  if (!src.includes('// Holocron Hot Module')) {
+    const modifiedSource = src.split('\n').map((line) => {
+      if (line.startsWith('export default')) {
+        const varName = line.replace('export default', '').replace(';', '').trim();
+
+        return [
+          'import { registerModule } from \'holocron\';',
+          `const HotHolocronModule = hot(${varName});`,
+          `registerModule('${moduleName}', HotHolocronModule);`,
+          'export default HotHolocronModule',
+        ].join('\n');
+      }
+      return line;
+    });
+    modifiedSource.unshift('import { hot } from \'react-hot-loader/root\';');
```

Comment:
The reason for a loader being added is that we need to update the source directly and inject imports.